### ISSUE TITLE
[Bug]: AbortController causes isSubmitting to be wrongly set to false

### ISSUE THREAD

**Issue by eMarek** [react-hook-form/react-hook-form#6619]
### Version Number

7.15.2

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-is-submitting-01n7c?file=/src/index.jsx:520-535

### Steps to reproduce

1. Go to provided Codesandbox. It has one input and submit button.
2. Open development tools in Chrome and throttle your network to something really slow (e.g. Slow 3G).

<img width="842" alt="Screenshot 2021-09-22 at 20 38 58" src="https://user-images.githubusercontent.com/311634/134402833-c6214178-5484-4ed5-9632-83aa6e07e85f.png">

3. Now fill the `Pet ID` input with number `1` and click `Submit` button. Nice `Loading...` message will appear below the button.
4. Wait 1 second (no more) and click `Submit` button again. At this point previous request is canceled and new is initialised. However, `Loading...` message disappears, but if you check your network tab you will see that retrieving the data is in progress.
5. After few seconds you will even receive and see pet's name on your screen.

https://user-images.githubusercontent.com/311634/134404280-80dce04a-b47b-4d1b-b1c9-b0b90bddf61a.mov

Based on this example you can see that `isSubmitting` is not properly set when seconds request is triggered.

### Expected behaviour

Obviously `isSubmitting` should be `true` even when user clicks button for the second time.

Spot this code in `onSubmit` callback:
```
    if (fetchControler) {
      fetchControler.abort();
    }
    fetchControler = new AbortController();
```

Basically when user clicks the `Submit` button I abort the previous fetch. Most likely this thing interferes with the `isSubmitting` flag.

Why do I need to abort the fetch? In reality I have an endpoint which takes quite a lot of time to process everything and I **can not block** user from changing the form values and submit them again while previous request is still on going, because of some business rules.

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
/
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
thanks for the detailed issue report.

I think this is not within the scope for the `isSubmitting` detection at the hook form level. https://github.com/react-hook-form/react-hook-form/blob/master/src/logic/createFormControl.ts#L1025

Feel free to submit a feature request instead but I highly recommend hosting such a state at your app level.
